src/hierarchies/mfa_hierarchy.h

```diff
@@ -0,0 +1,123 @@
+#ifndef BAYESMIX_HIERARCHIES_MFA_HIERARCHY_H_
+#define BAYESMIX_HIERARCHIES_MFA_HIERARCHY_H_
+
+#include <google/protobuf/stubs/casts.h>
+
+#include <Eigen/Dense>
+#include <memory>
+#include <vector>
+
+#include "algorithm_state.pb.h"
+#include "base_hierarchy.h"
+#include "hierarchy_id.pb.h"
+#include "hierarchy_prior.pb.h"
+
+//! Mixture of Factor Analysers hierarchy for multivariate data.
+
+//! This class represents a hierarchical model where data are distributed
+
```

Comment:
Are names to be changed? We noticed letters are never used for variables and instead more descriptive names are always used for parameters